controller/src/main/java/org/jboss/as/controller/operations/global/ReadResourceHandler.java

```diff
@@ -500,6 +504,9 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
                 ModelNode value = ar.response;
                 if (!value.has(FAILURE_DESCRIPTION)) {
                     sortedAttributes.put(entry.getKey(), value.get(RESULT));
+                    if (value.hasDefined(RESPONSE_HEADERS)) {
```

Comment:
Since we are only interested in the warnings, you could change the if to be:
```
if (value.hasDefined(RESPONSE_HEADERS, WARNINGS)) {
....                        
}
```